packages/cli/src/ui/commands/setupGithubCommand.ts

```diff
@@ -68,37 +73,87 @@ export const setupGithubCommand: SlashCommand = {
+    try {
+      await fs.promises.mkdir(githubWorkflowsDir, { recursive: true });
+    } catch (_error) {
+      console.debug(
+        `Failed to create ${githubWorkflowsDir} directory:`,
+        _error,
+      );
+      throw new Error(
+        `Unable to create ${githubWorkflowsDir} directory. Do you have file permissions in the current directory?`,
+      );
+    }
+
+    // Download each workflow in parallel - there aren't enough files to warrant
+    // a full workerpool model here.
     const workflows = [
       'gemini-cli/gemini-cli.yml',
       'issue-triage/gemini-issue-automated-triage.yml',
       'issue-triage/gemini-issue-scheduled-triage.yml',
       'pr-review/gemini-pr-review.yml',
     ];
 
-    const commands = [];
-
-    // Ensure fast exit
-    commands.push(`set -eEuo pipefail`);
-
-    // Make the directory if it doesn't exist
-    commands.push(`mkdir -p "${gitRepoRoot}/.github/workflows"`);
-
+    const downloads = [];
     for (const workflow of workflows) {
-      const fileName = path.basename(workflow);
-      const curlCommand = buildCurlCommand(
-        `https://raw.githubusercontent.com/google-github-actions/run-gemini-cli/refs/tags/${releaseTag}/examples/workflows/${workflow}`,
-        [`--output "${gitRepoRoot}/.github/workflows/${fileName}"`],
+      downloads.push(
+        (async () => {
+          const endpoint = `https://raw.githubusercontent.com/google-github-actions/run-gemini-cli/refs/tags/${releaseTag}/examples/workflows/${workflow}`;
+          const response = await fetch(endpoint, {
+            method: 'GET',
+            dispatcher: proxy ? new ProxyAgent(proxy) : undefined,
+            signal: AbortSignal.any([
+              AbortSignal.timeout(30_000),
+              abortController.signal,
+            ]),
+          } as RequestInit);
+
+          if (!response.ok) {
+            throw new Error(
+              `Invalid response code downloading ${endpoint}: ${response.status} - ${response.statusText}`,
+            );
+          }
+          const body = response.body;
+          if (!body) {
+            throw new Error(
+              `Empty body while downloading ${endpoint}: ${response.status} - ${response.statusText}`,
+            );
+          }
+
+          const destination = path.resolve(
+            githubWorkflowsDir,
+            path.join(path.basename(workflow)),
+          );
+
+          const fileStream = fs.createWriteStream(destination, {
+            mode: 0o640,
+            flags: 'w',
+            flush: true,
+          });
+
+          await body.pipeTo(Writable.toWeb(fileStream));
+        })(),
       );
-      commands.push(curlCommand);
     }
 
-    const readmeUrl = `https://github.com/google-github-actions/run-gemini-cli/blob/${releaseTag}/README.md#quick-start`;
+    // Wait for all downloads to complete
+    try {
+      await Promise.all(downloads);
+    } catch (_error) {
+      abortController.abort();
+      throw _error;
```

Comment:
instead of re-throwing an error turn the catch block into a finally block so the error will still throw but the controller will abort too